sunpy/net/tests/test_helioviewer.py

```diff
@@ -73,6 +73,18 @@ def test_download_jp2(self, client):
                                        instrument='MDI', measurement='continuum')
         assert "2011_01_11__22_39_00_000__SOHO_MDI_MDI_continuum.jp2" in filepath
         os.remove(filepath)
+        
+    def test_partial_match(self, client):
+        """
+        Tests the function that helps to access the helioviewer source id 
+        with fewer keywords.
+        """
+        data_dict = client.data_sources
+        key_1 = ('SOHO', 'MDI', None, 'continuum')
+        source_id_1 = data_dict[key_1]
+        key_2 = ('SOHO', None, None, 'continuum')
+        source_id_2 = list(client._partial_match(key=key_2, dictionary=data_dict))
```

Comment:
Why do you use `list` here? Also why are these passed in as keywords?

Reply:
That is fine then. But why are these passed in as keywords?

Reply:
You mean why is the key passed as  keyword?

Reply:
Yes.

Reply:
Actually I forgot to remove it. I was debugging the code so I added it. I will remove it